### PR TITLE
OCPBUGS-35800: kubelet config controller does not always propogate tlsSecurityProfile

### PR DESCRIPTION
**Pull request by djoshy**
**- What I did**

- All subcontrollers within the kubelet-config-controller will now apply the `APIServer` object settings, including their bootstrap variants. If a bootstrap `APIServer` manifest is provided, the controllers will use that - otherwise it will default to the intermediate TLS profile.
- All the controllers will resync on updates to the `APIServer` object. The initial resync is done by the node config controller, which subsequently calls the kubelet-config and feature sub controllers.
- Also added some unit tests to verify the above behavior.

**- How to verify it**

1. Bring up a cluster with an install time `APIServer` manifest named "cluster". Ensure that this manifest has [the TLS profile you'd like to test](https://docs.openshift.com/container-platform/4.15/security/tls-security-profiles.html#tls-profiles-kubernetes-configuring_tls-security-profiles). 
2. Once the install is complete, observe the kubelet configs(located at `/etc/kubernetes/kubelet.conf`) on any node to ensure that it has the correct values for TLSCiphers and TLSMinVersion.
3. Switch between TLS profiles by updating the `APIServer` object named "cluster".
4. This will cause a new `MachineConfig` rollout. You can verify the TLS values on the nodes are as expected as the update rolls through the cluster.
5. Now, try applying a user defined `KubeletConfig` with TLS settings that are different from the ones in the `APIServer` object. This should cause another `MachineConfig` rollout to the pool it was targeted to. After the update, you should see the nodes in the pool with the TLS values specified by the new `KubeletConfig`(not the `APIServer` object).

**Things to note**

- The APIServer rejects the Modern profile at the moment, according to the docs.
- If no tlsSecurityProfile is provided, the MCO will default to Intermediate.

### CODE REVIEW

File: pkg/controller/kubelet-config/kubelet_config_controller.go [hunk 1]
```diff
@@ -202,6 +209,61 @@ func (ctrl *Controller) Run(workers int, stopCh <-chan struct{}) {
 
 	<-stopCh
 }
+func (ctrl *Controller) filterAPIServer(apiServer *configv1.APIServer) {
+	if apiServer.Name == "cluster" {
```
Comment (rphillips) [inline pkg/controller/kubelet-config/kubelet_config_controller.go:213]: reversing the logic here would remove the nested if.

```suggestion
	if apiServer.Name != "cluster" {
	  return
	}  
```

**Comment by djoshy** [comment]
/test e2e-gcp-op-techpreview

/hold 
holding for QE approval

**Comment by ptalgulk01** [comment]
Pre-merge verified:
For manual- installation  and added the 99_apiserver.yaml file before creating cluster
```
$ cat > openshift/99_apiserver.yaml
apiVersion: config.openshift.io/v1
kind: APIServer
metadata:
  name: cluster
spec:
  tlsSecurityProfile:
    old: {}
    type: Old
  audit:
    profile: Default
```
```
$ oc get node
NAME                                        STATUS   ROLES                  AGE   VERSION
ip-10-0-28-203.sa-east-1.compute.internal   Ready    control-plane,master   93m   v1.30.1+9798e19
ip-10-0-28-40.sa-east-1.compute.internal    Ready    worker                 81m   v1.30.1+9798e19
ip-10-0-44-194.sa-east-1.compute.internal   Ready    worker                 82m   v1.30.1+9798e19
ip-10-0-61-35.sa-east-1.compute.internal    Ready    control-plane,master   93m   v1.30.1+9798e19
ip-10-0-76-153.sa-east-1.compute.internal   Ready    worker                 82m   v1.30.1+9798e19
ip-10-0-85-131.sa-east-1.compute.internal   Ready    control-plane,master   93m   v1.30.1+9798e19
```
More detail steps about verification are provided here: [MCO-850](https://issues.redhat.com/browse/MCO-850#:~:text=Manually%20installed%20the%20cluster%20and%20added%20the%2099_apiserver.yaml%20file%20before%20creating%20cluster)

Check that applied config is configured  on the node too.
```
 $ oc debug node/ip-10-0-76-153.sa-east-1.compute.internal -- chroot /host cat /etc/kubernetes/kubelet.conf
tlsCipherSuites:
- TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
- TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
- TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
- TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
- TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256
- TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
- TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256
- TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256
- TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA
- TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA
- TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA
- TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA
- TLS_RSA_WITH_AES_128_GCM_SHA256
- TLS_RSA_WITH_AES_256_GCM_SHA384
- TLS_RSA_WITH_AES_128_CBC_SHA256
- TLS_RSA_WITH_AES_128_CBC_SHA
- TLS_RSA_WITH_AES_256_CBC_SHA
- TLS_RSA_WITH_3DES_EDE_CBC_SHA
tlsMinVersion: VersionTLS10
``` 
Switch the tlsSecurityProfile via apiserver
```
$ oc get apiservers cluster  -o yaml
apiVersion: config.openshift.io/v1
kind: APIServer
metadata:
  creationTimestamp: "2024-08-08T06:51:20Z"
  generation: 2
  name: cluster
  resourceVersion: "45244"
  uid: 4b776b26-cab8-40ed-9edc-d28f7df7dc89
spec:
  audit:
    profile: Default
  tlsSecurityProfile:
    intermediate: {}
    type: Intermediate
```
```
$ oc get po kube-rbac-proxy-crio-ip-10-0-28-203.sa-east-1.compute.internal machine-config-controller-57fb7566ff-k2thk -o yaml | grep -i tls
      - --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
      - --tls-min-version=VersionTLS12

    - --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
    - --tls-min-version=VersionTLS12

$ oc logs machine-config-server-6q6gz | grep -i tls
I0808 08:36:48.025754       1 start.go:51] Launching server with tls min version: VersionTLS12 & cipher suites [TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384 TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256 TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256]
```
Apply custom created kubelet config with TLS settings that are different from the the APIServer object.
```
apiVersion: machineconfiguration.openshift.io/v1
kind: KubeletConfig
metadata:
  name: set-kubelet-tls-security-profile
spec:
  tlsSecurityProfile:
    type: Custom 
    custom: 
      ciphers: 
      - ECDHE-ECDSA-CHACHA20-POLY1305
      - ECDHE-RSA-CHACHA20-POLY1305
      - ECDHE-RSA-AES128-GCM-SHA256
      - ECDHE-ECDSA-AES128-GCM-SHA256
      minTLSVersion: VersionTLS11
  machineConfigPoolSelector:
    matchLabels:
      pools.operator.machineconfiguration.openshift.io/master: ""

$ oc apply -f kubletconfig.yaml 
kubeletconfig.machineconfiguration.openshift.io/set-kubelet-tls-security-profile created
 
$ oc get kubeletconfigs 
NAME                               AGE
set-kubelet-tls-security-profile   13s
```
Wait for particular pool to be updated on  which kubelet config is applied
```
$  oc get mcp
NAME     CONFIG                                             UPDATED   UPDATING   DEGRADED   MACHINECOUNT   READYMACHINECOUNT   UPDATEDMACHINECOUNT   DEGRADEDMACHINECOUNT   AGE
master   rendered-master-ac022d4da191a5039d6d5d61a4ea9c8c   True      False      False      3              3                   3                     0                      116m
worker   rendered-worker-2362c76331294e6a2eaa105f9f5ac0db   True      False      False      3              3                   3                     0                      116m
```
Check the `/etc/kubernetes/kubelet.conf` on the node has the same tls setting applied via kubelet config and does not contain API server applied tls setting 
```
$ oc debug node/ip-10-0-28-203.sa-east-1.compute.internal -- chroot /host cat  /etc/kubernetes/kubelet.conf | grep -i tls
Starting pod/ip-10-0-28-203sa-east-1computeinternal-debug ...
To use host binaries, run `chroot /host`

Removing debug pod ...
serverTLSBootstrap: true
tlsCipherSuites:
- TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256
- TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
- TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
- TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
tlsMinVersion: VersionTLS11
```
The tls setting is not disturbed by kubelet config in pods we could see the same tls setting as that of API server
```
$ oc get po kube-rbac-proxy-crio-ip-10-0-28-203.sa-east-1.compute.internal machine-config-controller-57fb7566ff-l7h9t oyaml | grep -i tls
      - --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
      - --tls-min-version=VersionTLS12

    - --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
    - --tls-min-version=VersionTLS12

$ oc logs machine-config-server-6q6gz | grep -i tls
I0808 08:36:48.025754       1 start.go:51] Launching server with tls min version: VersionTLS12 & cipher suites [TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384 TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256 TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256]
```

**Comment by ptalgulk01** [comment]
Pre-merge Verified on 
GCP - private-templates/functionality-testing/aos-4_17/ipi-on-gcp/versioned-installer
SNO - private-templates/functionality-testing/aos-4_17/ipi-on-gcp/versioned-installer
Parameters - enable_spot_instance_workers: "no"
num_masters: 1
num_workers: 0
master_worker_AllInOne: "true"
Disconnected - private-templates/functionality-testing/aos-4_17/ipi-on-gcp/versioned-installer_customer_vpc-disconnected
Proxy - private-templates/functionality-testing/aos-4_17/ipi-on-gcp/versioned-installer_customer_vpc-http_proxy

Steps: 

By default the tlsSecurityProfile is Intermediate , same can be seen on node in `/etc/kubernetes/kubelet.conf` config 
```
$ oc debug node/ptalgulk-07dis-47szz-worker-b-czxxs -- chroot /host cat /etc/kubernetes/kubelet.conf
tlsCipherSuites:
- TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
- TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
- TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
- TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
- TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256
- TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
tlsMinVersion: VersionTLS12
```
```
$ oc get po machine-config-controller-c775d6887-rmxg8 -oyaml | grep -i tls
 --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
    - --tls-min-version=VersionTLS12

$ oc get po kube-rbac-proxy-crio-ptalgulk-07dis-47szz-master-0 -oyaml | grep -i tls
--tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
    - --tls-min-version=VersionTLS12

$ oc logs machine-config-server-5wkzr | grep -i tls
I0807 03:28:58.175849       1 start.go:51] Launching server with tls min version: VersionTLS12 & cipher suites [TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384 TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256 TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256]
```
More detail steps about verification are provided here: [MCO-850](https://issues.redhat.com/browse/MCO-850#:~:text=Manually%20installed%20the%20cluster%20and%20added%20the%2099_apiserver.yaml%20file%20before%20creating%20cluster)

Switch  tlsSecurityProfile and wait till MCP is updated
Verify the tlsProfile is applied correctly 

```
$ oc debug node/ptalgulk-07dis-47szz-worker-b-czxxs -- chroot /host cat /etc/kubernetes/kubelet.conf
tlsCipherSuites:
- TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256
- TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
- TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
- TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
tlsMinVersion: VersionTLS11

$ oc get po machine-config-controller-bbd9fbd6b-7gbg9  -o yaml | grep -i tls
    - --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
    - --tls-min-version=VersionTLS11

$ oc logs machine-config-server-6rf7p | grep -i tls
I0807 05:11:03.628870       1 start.go:51] Launching server with tls min version: VersionTLS11 & cipher suites [TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256 TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256]

$ oc get po kube-rbac-proxy-crio-ptalgulk-07dis-47szz-master-0 -oyaml | grep -i tls
    - --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
    - --tls-min-version=VersionTLS11
```
```
$ oc debug node/ptalgulk-07dis-47szz-worker-b-czxxs  --image=quay.io/openshifttest/testssl@sha256:ad6fb8002cb9cfce3ddc8829fd6e7e0d997aeb1faf972650f3e5d7603f90c6ef -- testssl.sh --color 0 -E localhost:9637
TLS 1
 -
TLS 1.1
 -
TLS 1.2
 xcca9   ECDHE-ECDSA-CHACHA20-POLY1305     ECDH 253   ChaCha20    256      TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256      
 xc02b   ECDHE-ECDSA-AES128-GCM-SHA256     ECDH 253   AESGCM      128      TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256            
TLS 1.3
 x1302   TLS_AES_256_GCM_SHA384            ECDH 253   AESGCM      256      TLS_AES_256_GCM_SHA384                             
 x1303   TLS_CHACHA20_POLY1305_SHA256      ECDH 253   ChaCha20    256      TLS_CHACHA20_POLY1305_SHA256                       
 x1301   TLS_AES_128_GCM_SHA256            ECDH 253   AESGCM      128      TLS_AES_128_GCM_SHA256   
```

Apply kubelet config
```
apiVersion: machineconfiguration.openshift.io/v1
kind: KubeletConfig
metadata:
  name: set-kubelet-tls-security-profile
spec:
  tlsSecurityProfile:
    old: {}
    type: Old
  machineConfigPoolSelector:
    matchLabels:
      pools.operator.machineconfiguration.openshift.io/worker: ""

$ oc apply -f kubletconfig.yaml 
kubeletconfig.machineconfiguration.openshift.io/set-kubelet-tls-security-profile created
```

Wait for MCP to be updated on which kubelet is applied
```
$ oc debug node/ptalgulk-07dis-47szz-worker-a-r9gsg -- chroot /host cat /etc/kubernetes/kubelet.conf
tlsCipherSuites:
- TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
- TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
- TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
- TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
- TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256
- TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
- TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256
- TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256
- TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA
- TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA
- TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA
- TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA
- TLS_RSA_WITH_AES_128_GCM_SHA256
- TLS_RSA_WITH_AES_256_GCM_SHA384
- TLS_RSA_WITH_AES_128_CBC_SHA256
- TLS_RSA_WITH_AES_128_CBC_SHA
- TLS_RSA_WITH_AES_256_CBC_SHA
- TLS_RSA_WITH_3DES_EDE_CBC_SHA
tlsMinVersion: VersionTLS10
```
Kubelet Config does not affect the apiserver configuration applied other than on particular pool node

```
$ oc get po kube-rbac-proxy-crio-ptalgulk-07dis-47szz-worker-a-r9gsg -o yaml | grep -i tls
   --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
    - --tls-min-version=VersionTLS11

$ oc logs machine-config-server-6rf7p | grep -i tls
I0807 05:11:03.628870       1 start.go:51] Launching server with tls min version: VersionTLS11 & cipher suites [TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256 TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256]

$ oc debug node/ptalgulk-07dis-47szz-master-2 -- chroot /host cat /etc/kubernetes/kubelet.conf
tlsCipherSuites:
- TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256
- TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
- TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
- TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
tlsMinVersion: VersionTLS11
```

If we apply the tls setting after the kubelet config is applied it wont override the tls setting of kubelet config

e.g in SNO cluster

```
$ oc get apiservers.config.openshift.io cluster -o yaml
apiVersion: config.openshift.io/v1
kind: APIServer
….
  audit:
    profile: Default

$ oc get kubeletconfigs.machineconfiguration.openshift.io set-kubelet-tls-security-profile -o yaml
apiVersion: machineconfiguration.openshift.io/v1
kind: KubeletConfig
….
spec:
  machineConfigPoolSelector:
    matchLabels:
      pools.operator.machineconfiguration.openshift.io/master: ""
  tlsSecurityProfile:
    old: {}
    type: Old
…

$ oc debug node/ptalgulk-08sno-wj5bs-master-0   -- chroot /host cat /etc/kubernetes/kubelet.conf
tlsCipherSuites:
- TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
- TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
- TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
- TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
- TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256
- TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
- TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256
- TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256
- TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA
- TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA
- TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA
- TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA
- TLS_RSA_WITH_AES_128_GCM_SHA256
- TLS_RSA_WITH_AES_256_GCM_SHA384
- TLS_RSA_WITH_AES_128_CBC_SHA256
- TLS_RSA_WITH_AES_128_CBC_SHA
- TLS_RSA_WITH_AES_256_CBC_SHA
- TLS_RSA_WITH_3DES_EDE_CBC_SHA
tlsMinVersion: VersionTLS10
volumeStatsAggPeriod: 0s
```
Switch to Custom API server
```
$ oc get apiservers.config.openshift.io cluster -o yaml
….
  tlsSecurityProfile:
    custom:
      ciphers:
      - ECDHE-ECDSA-CHACHA20-POLY1305
      - ECDHE-RSA-CHACHA20-POLY1305
      - ECDHE-RSA-AES128-GCM-SHA256
      - ECDHE-ECDSA-AES128-GCM-SHA256
      minTLSVersion: VersionTLS11
    type: Custom

$  oc get po kube-rbac-proxy-crio-ptalgulk-08sno-wj5bs-master-0 -o yaml | grep -i tls
    - --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
    - --tls-min-version=VersionTLS11

```
Delete the Kubelet Config to have same tls setting as of API server

```
$ oc delete kubeletconfigs.machineconfiguration.openshift.io set-kubelet-tls-security-profile 
kubeletconfig.machineconfiguration.openshift.io "set-kubelet-tls-security-profile" deleted

$  oc debug node/ptalgulk-08sno-wj5bs-master-0 -- chroot /host cat /etc/kubernetes/kubelet.conf
tlsCipherSuites:
- TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256
- TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
- TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
- TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
tlsMinVersion: VersionTLS11

```

**Pull request merged**